mirgecom/simutil.py

```diff
@@ -943,87 +956,114 @@ def partition_generator_func(mesh, tag_to_elements, num_ranks):
-                for part_index, part_id in enumerate(part_id_to_elements.keys())}
-            from meshmode.mesh.processing import \
-                _compute_global_elem_to_part_elem
-            global_elem_to_part_elem = _compute_global_elem_to_part_elem(
-                mesh.nelements, part_id_to_elements, part_id_to_part_index,
-                mesh.element_id_dtype)
-
-            tag_to_global_to_part = {
-                tag: global_elem_to_part_elem[elements, :]
-                for tag, elements in tag_to_elements.items()}
-
-            part_id_to_tag_to_elements = {}
-            for part_id in part_id_to_elements.keys():
-                part_idx = part_id_to_part_index[part_id]
-                part_tag_to_elements = {}
-                for tag, global_to_part in tag_to_global_to_part.items():
-                    part_tag_to_elements[tag] = global_to_part[
-                        global_to_part[:, 0] == part_idx, 1]
-                part_id_to_tag_to_elements[part_id] = part_tag_to_elements
-
-            part_id_to_mesh = partition_mesh(mesh, part_id_to_elements)
-
-            rank_to_mesh_data = [
-                {
-                    vol: (
-                        part_id_to_mesh[PartID(vol, rank)],
-                        part_id_to_tag_to_elements[PartID(vol, rank)])
-                    for vol in volumes}
-                for rank in range(num_ranks)]
-
-        local_mesh_data = comm.scatter(rank_to_mesh_data, root=0)
-
-        global_nelements = comm.bcast(mesh.nelements, root=0)
+                    for vol_idx in range(len(volumes))
+                    for rank in range(num_ranks)}
+
+                # TODO: Add a public meshmode function to accomplish this? So we're
+                # not depending on meshmode internals
+                part_id_to_part_index = {
+                    part_id: part_index
+                    for part_index, part_id in enumerate(part_id_to_elements.keys())}
+                from meshmode.mesh.processing import \
+                    _compute_global_elem_to_part_elem
+                global_elem_to_part_elem = _compute_global_elem_to_part_elem(
+                    mesh.nelements, part_id_to_elements, part_id_to_part_index,
+                    mesh.element_id_dtype)
+
+                tag_to_global_to_part = {
+                    tag: global_elem_to_part_elem[elements, :]
+                    for tag, elements in tag_to_elements.items()}
+
+                part_id_to_tag_to_elements = {}
+                for part_id in part_id_to_elements.keys():
+                    part_idx = part_id_to_part_index[part_id]
+                    part_tag_to_elements = {}
+                    for tag, global_to_part in tag_to_global_to_part.items():
+                        part_tag_to_elements[tag] = global_to_part[
+                            global_to_part[:, 0] == part_idx, 1]
+                        part_id_to_tag_to_elements[part_id] = part_tag_to_elements
```

Comment:
@MTCam Just noticed this while comparing new/old code in #968: I think the line here got indented one too many times. It should be outside of the `tag_to_global_to_part` loop I believe.